ci/test_cudf_polars.sh

```diff
@@ -0,0 +1,54 @@
+#!/bin/bash
+# Copyright (c) 2024, NVIDIA CORPORATION.
+
+set -eou pipefail
+# TODO: is the target branch exposed anywhere in an environment variable?
+git fetch origin branch-24.08
+
+# We will only fail these tests if the diff touches code in pylibcudf or cudf_polars itself.
+if [ -n "$(git diff --name-only branch-24.08..HEAD -- python/cudf_polars/ python/cudf/cudf/_lib/pylibcudf/)" ];
+then
+    HAS_CHANGES=1
+else
+    HAS_CHANGES=0
+fi
+
+RAPIDS_PY_CUDA_SUFFIX="$(rapids-wheel-ctk-name-gen ${RAPIDS_CUDA_VERSION})"
+RAPIDS_PY_WHEEL_NAME="cudf_${RAPIDS_PY_CUDA_SUFFIX}" rapids-download-wheels-from-s3 ./dist
+
+rapids-logger "Install cudf wheel"
+# echo to expand wildcard before adding `[extra]` requires for pip
+python -m pip install $(echo ./dist/cudf*.whl)[test]
+
+rapids-logger "Install polars (allow pre-release versions)"
+python -m pip install --pre polars
```

Comment:
Passing `--pre` means that `pip install` will also install dev releases of `polars` dependencies, which you might not want.

It'd be better to express the requirement like this:

```text
polars>=1.0.0a0
```

In the current state of `polars` on PyPI, that'll pull in `1.0.0.rc2`. You can test like this:

```shell
pip install --dry-run 'polars>=1.0.0a0'
```

The trick here is that if you provide any dev specifiers (like that version with an `a0`), `pip` will consider dev releases **of that specific requirement** but not any of its dependencies.

Doing that would allow you to remove this extra `pip install` in a script and have this dependency flow through the package metadata.

> *you can put pip dependencies/pip flags there*

Just to clarify... the `pip` flags you can include are only those that'd be valid in a `requirements.txt` file. The list of those options is described here: https://pip.pypa.io/en/stable/reference/requirements-file-format/#supported-options